packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -14,6 +14,27 @@ enum ViewportAnchor {
   end,
 }
 
+class ViewportDimensions {
+  const ViewportDimensions({
+    this.contentSize: Size.zero,
+    this.containerSize: Size.zero
+  });
+
+  static const ViewportDimensions zero = const ViewportDimensions();
+
+  final Size contentSize;
+  final Size containerSize;
+
+  Offset getAbsolutePaintOffset({ Offset paintOffset, ViewportAnchor anchor }) {
+    switch (anchor) {
```

Comment:
assert that at least one of the axes has the same dimension in the content and container